### PR TITLE
Registry Sync and Registering, attempt 2

### PR DESCRIPTION
**Pull request by BoogieMonster1O1**
Goals: 
- [x] Sync Block and Item IDs between the server and client.
- [x] Save Block and Item IDs to a file to prevent changes in IDs on restart.
- [x] Allow easier registration of Blocks and Items

### CODE REVIEW

**Comment by BoogieMonster1O1** [comment]
Please review thoroughly, and leave comments on how the implementation for registry remapping can be improved.

**Review by arthurbambou — APPROVED** [review]
It looks good globally except the fact there are 3 remapper instance per registry which looks somewhat odd.

**Review by PseudoDistant — APPROVED** [review]
The

Now we need to backport Indigo.

**Comment by arthurbambou** [comment]
merge upstream into your branch, it should fix the crash you have.

File: legacy-fabric-registry-sync-api-v1/build.gradle [hunk 1]
```diff
@@ -0,0 +1,5 @@
+version = getSubprojectVersion(project, "1.0.0")
+
+dependencies {
+	implementation project(path: ':legacy-fabric-api-base', configuration: 'dev')
```
Comment (arthurbambou) [inline legacy-fabric-registry-sync-api-v1/build.gradle:4]:
```suggestion
	implementation project(path: ':legacy-fabric-api-base', configuration: 'dev')
	implementation project(path: ':legacy-fabric-networking-api-v1', configuration: 'dev')
```
This should fix the build I guess

File: legacy-fabric-registry-sync-api-v1/src/main/java/net/legacyfabric/fabric/impl/registry/sync/RegistryRemapper.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.legacyfabric.fabric.impl.registry.sync;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
```
Comment (arthurbambou) [inline legacy-fabric-registry-sync-api-v1/src/main/java/net/legacyfabric/fabric/impl/registry/sync/RegistryRemapper.java:21]:
```suggestion
```

File: legacy-fabric-registry-sync-api-v1/src/main/java/net/legacyfabric/fabric/impl/registry/sync/RegistryRemapper.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+import java.util.function.IntSupplier;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.PacketByteBuf;
```
Comment (arthurbambou) [inline legacy-fabric-registry-sync-api-v1/src/main/java/net/legacyfabric/fabric/impl/registry/sync/RegistryRemapper.java:32]:
```suggestion
```

**Review by arthurbambou — COMMENTED** [review]
styling

**Review by arthurbambou — APPROVED** [review]

**Pull request merged**